lib/relation-definition.js

```diff
@@ -1943,9 +1985,9 @@ HasOne.prototype.getAsync = function(options, cb) {
   return cb.promise;
 };
 
-RelationDefinition.embedsOne = function(modelFrom, modelTo, params) {
+RelationDefinition.embedsOne = function(modelFrom, modelToRef, params) {
   params = params || {};
-  modelTo = lookupModelTo(modelFrom, modelTo, params);
+  var modelTo = lookupModelTo(modelFrom, modelToRef, params);
 
   var thisClassName = modelFrom.modelName;
   var relationName = params.as || (i8n.camelize(modelTo.modelName, true) + 'Item');
```

Comment:
same here: `params.reference || params.as || i8n.camelize(modelTo.pluralModelName, true);` if we decide to use reference